applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_process_utilities.cpp

```diff
@@ -141,4 +150,95 @@ static const std::vector<NamedFactory> ProcessesAndOperations = {
+        {
+            "model_part_name": "Main"
+        })");
+
+    // Act
+    ProcessUtilities::AddProcessesSubModelPartList(project_parameters, solver_settings);
+
+    // Assert
+    KRATOS_EXPECT_TRUE(solver_settings.Has("processes_sub_model_part_list"))
+    KRATOS_EXPECT_TRUE(solver_settings["processes_sub_model_part_list"].IsArray())
+    KRATOS_EXPECT_TRUE(solver_settings["processes_sub_model_part_list"].size() == 0)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(AddProcessesSubModelPartList_AddsFilledList, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Set
+    const Parameters project_parameters(R"(
+      {
+         "processes":
+         {
+            "constraints_process_list": [
+            {
+                "Parameters":    {
+                    "model_part_name": "PorousDomain.Sides"
+                }
+            },{
+                "Parameters":    {
+                    "model_part_name": "PorousDomain.BottomFixed"
+                }
+            },{
+                "Parameters":    {
+                   "model_part_name": "PorousDomain.TopLoad"
+                }
+            }],
+            "loads_process_list": [
+            {
+                "Parameters":    {
+                    "model_part_name": "PorousDomain.Soil"
+                }
+            },{
+                "Parameters":    {
+                   "model_part_name": "PorousDomain"
+                }
+            }],
+            "auxiliary_process_list": [{
+                "Parameters": {
+                    "model_part_name": "PorousDomain.Soil"
+                }
+            }],
+            "arbitrary_process_list": [{
+                "Parameters": {
+                    "model_part_name": "PorousDomain.Domain"
+                }
+            }]
+         },
+         "solver_settings": {
+             "model_part_name": "PorousDomain",
+             "processes_sub_model_part_list": ["name1", "name2", "name3"]
+         }
+      })");
+
+    auto solver_settings = project_parameters["solver_settings"];
+
+    std::stringstream buffer;
+    std::streambuf*   old = std::cout.rdbuf(buffer.rdbuf()); // NOSONAR
+
+    // Act
+    ProcessUtilities::AddProcessesSubModelPartList(project_parameters, solver_settings);
+
+    // Assert
+    std::cout.rdbuf(old); // NOSONAR
+    std::string output = buffer.str();
+    EXPECT_NE(output.find("processes_sub_model_part_list is deprecated"), std::string::npos);
+
+    const auto& r_list = solver_settings["processes_sub_model_part_list"];
+    KRATOS_EXPECT_EQ(r_list.size(), 4);
+
+    std::set<std::string, std::less<>> actual_modelpart_names;
+    for (const auto& r_name : r_list) {
+        actual_modelpart_names.insert(r_name.GetString());
+    }
+    const std::set<std::string, std::less<>> expected_modelpart_names = {"Sides", "BottomFixed",
+                                                                         "TopLoad", "Soil"};
+    EXPECT_EQ(actual_modelpart_names, expected_modelpart_names);
```

Comment:
I think we should use std::vector as container here, because std::set already sorts and removes duplicates (meaning we don't test if duplicates are correctly handled in the function)
```suggestion
    std::vector<std::string> actual_modelpart_names;
    for (const auto& r_name : r_list) {
        actual_modelpart_names.push_back(r_name.GetString());
    }
    const std::vector<std::string> expected_modelpart_names = {"Sides", "BottomFixed",
                                                                         "TopLoad", "Soil"};
    EXPECT_EQ(actual_modelpart_names, expected_modelpart_names);
```

Reply:
Addition: I see that you check the size of the list to be correct at the start, so you actually already check implicitly there are no duplicates. Still I think it's better to use std::vector instead of std::set

Reply:
changed to vector.